### DISCUSSION TITLE
Data Calibration and Experimental Signal Inversion

### DISCUSSION THREAD

**Discussion by tjk9501** [slimgroup/JUDI.jl#101] [Q&A] [answered]
Hello, Dr.Louboutin!

During the use of Devito and JUDI package, I confronted with the following two problems:
1. If we get some experimental time signals from equipments, how can we save the experimental data into the specific SEGY format which can be directly used by JUDI package?
2. As we know, the authentic experimental data will eventually deviate from time signals obtained from JUDI's acoustic modeling (like both amplitude and phase deviations), and it seems that JUDI uses time signals from FWI inversion. How to calibrate the experimental time signals so that during the JUDI's FWI process we can avoid divergence results and achieve better accuracy?

Thanks!

**Comment by mloubout** [comment, marked as answer]
Hi, thanks for the interest. Let me try to answer both points.

1. Unfortunatley, this is something you will have to check with your equipment provider. However, SEGY is no mandatory. If you check https://github.com/slimgroup/JUDI.jl/discussions/97 you'll see that you can work with your own data format with a little rapper to define your own `judiVector`. So  I would recommend to work with whatever data format is the easiest for you.
2. Could you expand on this point, I am not sure I understand the question.

**Reply by tjk9501** [reply]
Thanks! Let me clarify my problems.

1. Assuming that I deploy 16 sensors on a small Marmousi model, in the experiment I set the time sampling spacing and sampling length to be 0.08us and 160us. Then I save my data into a 16 by 16 by 2001 three dimensional array. So, where can I find the documentations to tell me what I shall do (use MATLAB or PYTHON external packages) to store this 3D array into the specific SEGY format that can be directly used by JUDI and DEVITO?
2. Let's also take the aforementioned case as the example. For instance, the simulation time signals I obtained from JUDI are d1 and the signals I obtained from experiment (with same setup as simulation) are d2. In most cases, the wavepackets in d2 will have amplitudes attenuation and arrive sooner or later than d1 from 3D acoutic modeling. Now I want to minimize these differences (preprocessing time signals of d2, I call it calibration) so that after I used d2 as observed data I will not get diverged results with FWI. Are there any suggested efficient way to minimize the differences  between d1 and d2 with JUDI?
3. I want to modify some of provided codes in JUDI so that I can use SGD or other methods in 3D FWI, but after I installed SlimOptim I found out backtracking_linesearch function does not exist any more. The closet candidate is BackTracking funtion, but it cannot be used as the FWI 2d example said. Which package shall I install to obtain the 3D backtracking_linesearch function (it seems that the provided 3D SPG FWI example also cannot be used anymore)?
PS: I am using Julia 1.15 and installed the JUDI package in July of 2021, I didn't update JUDI because I heard from friends after updates some of the previous FWI functions are wasted.
Thanks!

**Reply by mloubout** [reply]
1. JUDI accept pure vectors as input as well as long as you provide the geometry so you can just load the data in a vector and use it directly
2. JUDI does not provide this type of data processing tools but only the inversion part. So you would have to preprocess the data `d2` before using it in JUDI
3.  `backtracking_linesearch` was a homemade linesearch function we deprecated years ago, it is not part of LineSearches.jl . All the examples in JUDI are all tested and ran as part of CI so there is no issue with them. So you can use any of the FWI examples. You just need to update JUDI and the examples. There hasn't been any breaking changes in JUDI in quite a while, the last time was when as yu pointed out we created SLimOptim which was almost two years ago.

I would really recommend to upgrade your julia version as well. Julia now has switched to `1.6` as the long time support version meaning they do not officially maintain and support version older than that.


Hope this help and I'll gladly guide you through the examples so that you can be setup
